File: Node.java
public class Node<T> {
	
	private T value;
	private T right = null;
	private T left = null;
	
	public Node(T value) {
		this.value = value;
	}
	
	public Node(T value, T right) {
		this.value = value;
		this.right = right;
	}
	
	public Node(T value, T right, T left) {
		this.value = value;
		this.right = right;
		this.left = left;
	}

	public T getValue() {
		return this.value;
	}

	public void setValue(T value) {
		this.value = value;
	}

	public T getRight() {
		return this.right;
	}

	public void setRight(T right) {
		this.right = right;
	}

	public T getLeft() {
		return this.left;
	}

	public void setLeft(T left) {
		this.left = left;
	}
}

File: Node.js
class node {
    constructor(left, right, value) {
        this.left = left
        this.right = right
        this.value = value
    }

    getValue() {
        return this.value
    }

    getRight() {
        return this.right
    }

    getLeft() {
        return this.left
    }

    hasValue() {
        return this.value != undefined
    }

    hasRight() {
        return this.right != undefined
    }

    hasLeft() {
        return this.left != undefined
    }

    setRight(right) {
        this.right = right
    }

    setLeft(left) {
        this.left = left
    }

    setValue(value) {
        this.value =value
    }
}
